src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -376,7 +377,9 @@ def name(self) -> str:
 
     @property
     def version(self) -> CandidateVersion:
-        return self.dist.version
+        if self._version is None:
+            self._version = self.dist.version
+        return self._version
```

Comment:
One could use `cached_property` as well: https://docs.python.org/3/library/functools.html?highlight=cached_property#functools.cached_property

i.e.

```python
from functools import cached_property

...

    @cached_property
    def version(self) -> CandidateVersion:
        return self.dist.version
```

So we can also remove the `self._version` on `__init__` method 👍🏾

Reply:
It’s not available in python 3.7, unfortunately